storage/remote/otlptranslator/prometheus/normalize_name.go

```diff
@@ -125,7 +129,7 @@ func normalizeName(metric pmetric.Metric, namespace string) string {
 	if len(unitTokens) > 0 {
 		mainUnitOTel := strings.TrimSpace(unitTokens[0])
 		if mainUnitOTel != "" && !strings.ContainsAny(mainUnitOTel, "{}") {
-			mainUnitProm := CleanUpString(unitMapGetOrDefault(mainUnitOTel))
+			mainUnitProm := CleanUpString(unitMapGetOrDefault(mainUnitOTel), allowUTF8)
```

Comment:
Having thought on it some more, I think it's an anti-pattern to add the `allowUTF8` parameter to `CleanUpString`. When `allowUTF8 == true`, it's a no-op so it's pointless.
```suggestion
			mainUnitProm := unitMapGetOrDefault(mainUnitOTel)
			if !allowUTF8 {
				mainUnitProm = CleanUpString(mainUnitProm)
			}
```